cli/src/pcluster/validators/ec2_validators.py

```diff
@@ -47,13 +50,13 @@ class InstanceTypeBaseAMICompatibleValidator(Validator):
 
     def _validate(self, instance_type: Param, parent_image: Param):
```

Comment:
It seems to me that we're mixing the validators for imagebuilder and cluster.

If they are very imagebuilder specific let's create a imagebuilder_validator module (e.g. if they are using ImageBuilder boto3 calls inside).

Reply:
I am thinking we can share this validator with Cluster and Imagebuilder. I just need to rename parent_image to image, and `get_ami_id` won't call ImageBuilder boto3 calls if `image` is an ami_id instead of image_arn.